app/server/index.js

```diff
@@ -64,25 +64,52 @@ app.prepare().then(() => {
       classicIds: true,
       pgSettings(req) {
         const claims = {};
-        if ((((req.kauth || {}).grant || {}).id_token || {}).content) {
-          // TODO: actually map jwt realms to postgres roles
-          // @see https://www.postgresql.org/docs/current/default-roles.html
-          // claims['role'] = 'pg_monitor';
-          const token = req.kauth.grant.id_token.content;
-          for (const property in token) {
-            claims[`jwt.claims.${property}`] = token[property];
-          }
-        }
-
-        console.dir(claims);
-        return {
-          ...claims
-        };
+        if (
+          !req.kauth ||
+          !req.kauth.grant ||
+          !req.kauth.grant.id_token ||
+          !req.kauth.grant.id_token.content
+        )
+          return claims;
+
+        // TODO: actually map jwt realms to postgres roles
+        // @see https://www.postgresql.org/docs/current/default-roles.html
+        // claims['role'] = 'pg_monitor';
+        const token = req.kauth.grant.id_token.content;
+        const properties = [
+          'jti',
+          'exp',
+          'nbf',
+          'iat',
+          'iss',
+          'aud',
+          'sub',
+          'typ',
+          'azp',
+          'auth_time',
+          'session_state',
+          'acr',
+          'email_verified',
+          'name',
+          'preferred_username',
+          'given_name',
+          'family_name',
+          'email'
+        ];
+        properties.forEach(property => {
+          claims[`jwt.claims.${property}`] = token[property];
+        });
+
+        return claims;
       }
     })
   );
 
-  server.get('/form-builder', keycloak.protect());
+  server.post('/login', keycloak.protect(), (req, res) =>
+    res.redirect(302, '/user-dashboard')
+  );
```

Comment:
@matthieu-foucault the 302 redirect here assumes the error case is handled by a correctly configured `keycloak.protect()` middleware and that there's either a login form embedded on the page, or a discrete login page with a form configured to produce this post request